### ISSUE TITLE
Hide or remove provider's secrets in the administration "system info" page

### ISSUE THREAD

**Issue by marcbria** [pkp/pkp-lib#10213]
**Describe the bug**

This issue is a generalization of #7716.

In short, every user with rights to see the Administration pages can see sensitive information (pwds, secrets, keys, credentials...).

Actually, why would an ojs administrator need access to this information? As far as sysadmins already know this (or can check it in the config.inc.php) may be is a good idea NOT showing this info in this page... or at least, transform it to asterisks. 

Finally, at the end of the page, there is a link to a `phpinfo()` that will also expose sensitive data.

A preliminary list of fields I suggest being removed (you can always check them in config.ini.php and been able to reach the file system is also a nice filter) would be:

- [x] database.password (done)
- [x] security.salt
- [x] security.apy_key_secret
- [x] email.smtp_user
- [x] email.smtp_password
- [x] email.smtp_oauth_clientsecret
- [x] captcha.recaptcha_private_key
- [ ] phpinfo()

In the former post, @ctgraham explains this in a more clear way:

> I make a distinction between credentials that I would pass along to a new owner, and those credentials which I would keep private, if I transferred the OJS installation to another party.
> From your list, I think the database password, the smtp secrets, and the ReCAPTCHA private key all fall into the category of secrets that belong to the hosting provider, not to the hosted journal.

**This thread is to talk and decide if something else need to be hidden... or we are fine.**

**To Reproduce**

1. Go to 'Administration'
2. Click on 'System Information' (url: https://foojournal.org/admin/systemInfo)
3. See the pwds, secrets, credentials and keys.

**What application are you using?**
OJS, 3.x

**Comment by taslangraham** [comment]
@asmecher I added PRs for the preliminary list mentioned in the description.
pkp-lib stable 3.4 - https://github.com/pkp/pkp-lib/pull/10236/files
pkp-lib  stable 3.3 - https://github.com/pkp/pkp-lib/pull/10238/files
pkp-lib main - https://github.com/pkp/pkp-lib/pull/10237/files

**Comment by asmecher** [comment]
@marcbria, I think being able to direct users to a `phpinfo()` page is a helpful thing to put behind an admin login; are you sure it's justified to remove it? (What's the particularly sensitive information you'd like to protect?)

@taslangraham, depending on the outcome of that conversation, you'll either need to restore the link to the `phpinfo()` output, or fully remove that function from the handler.

**Comment by marcbria** [comment]
Hi Alec,

TL;DR; Currently nothing dramatic is shown, but it is good security practice not to show more info than necessary and may be a problem in the near future as it shows environment variables.

I expand a bit:

Certainly the phpinfo thing is a bit to go by note (it's not like the secrets we expose on the info page that an OJS admin should not know), but there is some info that is better not to expose. I am referring to IPs, software versions and above all environment variables.

Currently there are not very procupying variables, but if at some point the support for envVARs is implemented (see #8015), in that section access credentials would be shown and that is again a risk.

If it were a resource widely used by admins, I would rethink the suggestion, but searching the forum, I don't see any reference to “admin/phpinfo” and all comments suggest creating a file on disk.

I actually see very few scenarios where an admin has access to the OJS admin user but does not have access to the file system and.... I also think it's good to keep a good separation between ojs-admin and sysadmin, as the pwd of the former are usually managed more happily than the latter.

In any case, if you consider that removing the link is a loss, I will not insist, because as I said at the beginning, it is more prophylactic than a real problem.

**Comment by ctgraham** [comment]
I like having the `phpinfo()` function available for the administrator(s), though I might change my mind if sensitive environment variables were present.  In my case, the system administrators (with access to the filesystem and OS) and the OJS administrators (with access to the admin credential) are not always the same, but are always trusted.  I can imagine a future scenario where I would want to give the OJS admin credential (for the purpose of creating new journals) to user who should not have access to the filesystem or OS.

For comparison on `phpinfo()`:
* In Drupal, embedding a `phpinfo()` link is standard practice.
* In Wordpress, `phpinfo()` is available via a plugin.

**Comment by marcbria** [comment]
I don't have a strong position with this so if you both agree, I'm also fine... at least, till we don't have the phpdotenv thing implemented.

@asmecher should we keep this post open to remember or it's better closing it?

**Comment by asmecher** [comment]
Let's keep the phpinfo output for now. @taslangraham, can you remove that part of the PRs?

**Comment by taslangraham** [comment]
> Let's keep the phpinfo output for now. @taslangraham, can you remove that part of the PRs?
@asmecher I've updated the PRs

**Comment by asmecher** [comment]
All merged -- thanks!
